src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Result/ArtworksRail.tsx

```diff
@@ -39,48 +41,84 @@ export const ArtworksRail: FC<ArtworksRailProps> = props => {
   const { state } = props
   const [artworks, setArtworks] = useState<DiscoveryArtwork[]>([])
   const [isLoading, setIsLoading] = useState<boolean>(true)
+  const [excludeArtworkIds, setExcludeArtworkIds] = useState<string[]>([])
+
+  const { user } = useSystemContext()
 
   useEffect(() => {
-    const params = new URLSearchParams()
-    state.interests.forEach(concept => {
-      params.append("concepts", concept)
-    })
+    const fetchArtworks = async (options: {
+      concepts: string[]
+      excludeArtworkIds: string[]
+    }) => {
+      setIsLoading(true)
+
+      const { concepts, excludeArtworkIds } = options
+      const { priceMinUSD, priceMaxUSD } = getPriceRange(state.budgetIntent)
 
-    const { priceMinUSD, priceMaxUSD } = getPriceRange(state.budgetIntent)
-    if (priceMinUSD) params.append("priceMinUSD", priceMinUSD.toString())
-    if (priceMaxUSD) params.append("priceMaxUSD", priceMaxUSD.toString())
+      const params = new URLSearchParams()
+
+      if (priceMinUSD) params.append("priceMinUSD", priceMinUSD.toString())
+      if (priceMaxUSD) params.append("priceMaxUSD", priceMaxUSD.toString())
+      concepts.forEach(concept => {
+        params.append("concepts", concept)
+      })
+      excludeArtworkIds.forEach(id => {
+        params.append("excludeArtworkIds", id)
+      })
+      params.append("userId", user?.id || "")
+      params.append("goal", state.goal)
 
-    const fetchArtworks = async () => {
       const response = await fetch(
         `/api/advisor/7/artworks?${params.toString()}`
       )
       const data = await response.json()
-      setArtworks(data)
       setIsLoading(false)
+      return data
     }
 
-    fetchArtworks()
-  }, [state.interests, state.goal, state.budgetIntent])
+    const options = {
+      concepts: state.interests,
+      userId: user?.id,
+      excludeArtworkIds,
+    }
 
-  if (isLoading) {
-    return <Text>Loading...</Text>
-  }
+    fetchArtworks(options).then(setArtworks)
+  }, [state.interests, state.goal, state.budgetIntent, excludeArtworkIds, user])
 
   return (
     <>
-      {artworks.length ? (
+      {isLoading && artworks.length ? (
+        <Skeleton>
+          <Shelf>
+            {[...new Array(8)].map((_, i) => {
+              return (
+                <ShelfArtworkPlaceholder
+                  key={i}
+                  index={i}
+                  hideSaleInfo={true}
+                  maxImageHeight={400}
+                />
+              )
+            })}
+          </Shelf>
+        </Skeleton>
```

Comment:
Added this, which is the canonical rail loading screen. There is still a bit of ui shifting that needs to be addressed. But seemed reasonable to handle that better at the ui polish stage, if we keep this.